### PR TITLE
Fixes issue with debounced filter input

### LINKED ISSUE #7147: Filter value input bug
# Bug report

## Describe the bug

> I'm entering text in the table filter component and the filter component is deleting it as I type. Using Firefox 99.0 (64bit) Table Filter is unusable.

https://user-images.githubusercontent.com/689843/164185578-53f1d18a-6f91-4275-be7c-71b536d2bbdc.mov

### PR DESCRIPTION
**Pull request by saltcod**
Resolves https://github.com/supabase/supabase/issues/7147

@phamhieu if you could have a look at this one please. 

Having `now` in the effect is causing the component to constantly rerender. I'm trying to figure out why `now` was there in the first place. The comment suggests that the query won't update when you delete a filter row, but it does seem to do that here. 

Wondering if I'm missing something. 

https://user-images.githubusercontent.com/105593/172445459-7f99f107-0ed5-450e-899d-8d4398a41aae.mp4

### CODE REVIEW

**Comment by Ziinc** [comment]
@saltcod had a look at the code. The comment is correct in that the way that the `key` is set up means that filters after the deleted row will not be rerendered properly, because they will shift up in the array and have the same key as the deleted row, resulting in the deleted row still being shown.

It looks ok in your demo because you are deleting the last item, resulting in array size changing from 2 to 1, and last index item getting deleted. First index row is unchanged, hence not rerendered. 

You can fix this by generating a hash or id on filter row creation, and using an object to store the filters instead of an array.

File: studio/components/grid/components/header/filter/FilterPopover.tsx [hunk 1]
```diff
@@ -48,7 +48,7 @@ const Filter: FC = () => {
     <div className="space-y-2 py-2">
       <div className="space-y-2">
         {state.filters.map((_, index) => (
```
Comment (Ziinc) [inline studio/components/grid/components/header/filter/FilterPopover.tsx:50]: should not use array index for identifying the row for rerenders. can replace with a string based on filter value.

**Comment by joshenlim** [comment]
quick observation - the input field loses focus after the debounce which is not the best UX though if we can find a fix for that

**Comment by joshenlim** [comment]
just digging into this - the issue with the wrong behaviour when deleting filters seem to stem from the reducer itself

**Comment by joshenlim** [comment]
Added a change - assigning a uuid to each filter which should be better as a `key` than concatening the various parts of the filter - seems to resolve the focus lost issue too

**Review by joshenlim — APPROVED** [review]

**Pull request merged**